torch/onnx/symbolic_opset9.py

```diff
@@ -1903,21 +1903,28 @@ def new_full(g, self, size, fill_value, dtype, layout, device, pin_memory=False)
 
 
 def eye(g, *args):
-    if len(args) == 5:
-        # aten::eye(n, dtype, layout, device, pin_memory)
-        n, dtype, layout, device, pin_memory = args
+    n, m, k, dtype, layout, device, pin_memory = args
+    is_m_none = m.node().kind() == "prim::Constant" and m.type().kind() == "NoneType"
+    if (is_m_none):
         dim_size = sym_help._unsqueeze_helper(g, n, [0])
         shape = g.op("Concat", dim_size, dim_size, axis_i=0)
-        tensor = zeros(g, shape, dtype, layout, device)
-        return g.op("EyeLike", tensor)
-    elif len(args) == 6:
-        # aten::eye(n, m, dtype, layout, device, pin_memory)
-        n, m, dtype, layout, device, pin_memory = args
-        shape = g.op("Concat", sym_help._unsqueeze_helper(g, n, [0]), sym_help._unsqueeze_helper(g, m, [0]), axis_i=0)
-        tensor = zeros(g, shape, dtype, layout, device)
+    else:
+        n_size = sym_help._unsqueeze_helper(g, n, [0])
+        m_size = sym_help._unsqueeze_helper(g, m, [0])
+        shape = g.op("Concat", n_size, m_size, axis_i=0)
+    tensor = zeros(g, shape, dtype, layout, device)
+    is_k_none = k.node().kind() == "prim::Constant" and k.type().kind() == "NoneType"
+    if (is_k_none):
         return g.op("EyeLike", tensor)
     else:
```

Comment:
Reduce nesting of the if, and leave the error case for the end:
```suggestion
    elif k.node().kind() == "onnx::Constant":
       ...
    else:
      raise RuntimeError(...)
```